javaparser-core/src/main/java/com/github/javaparser/ast/Node.java

```diff
@@ -158,9 +203,11 @@ public final int getEndLine() {
 	 * 
 	 * @param beginColumn
 	 *            the begin column of this node
+	 * @deprecated Use setBegin()
 	 */
+	@Deprecated()
 	public final void setBeginColumn(final int beginColumn) {
-		this.beginColumn = beginColumn;
+		begin = new Position(begin == null ? 0 : begin.line, beginColumn);
```

Comment:
I would add a method to position which takes the column a return an instance of Position with the same line and the given column:
```
begin = begin.withColumn(beginColumn);
```